torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -339,6 +381,19 @@ def __init__(
             for p in self.params:
                 self._offload_to_cpu(p)
 
+    def _init_reshard_after_forward(self):
+        if self.sharding_strategy == ShardingStrategy.FULL_SHARD:
+            # Free full params and keep shard only after forward
+            self.reshard_after_forward = True
+        elif self.sharding_strategy == ShardingStrategy.SHARD_GRAD_OP:
+            # Keep full params in the GPU memory until backward
+            # computation is done
+            self.reshard_after_forward = False
+        else:
+            raise RuntimeError(
+                "sharding_strategy only supports FULL_SHARD and SHARD_GRAD_OP right now."
```

Comment:
nit: If we're raising an error here then we might not need to leave things like "# SHARD_OP = auto()" commented out since the user will get this unsupported error. But I think it's fine either way